mlflow/models/utils.py

```diff
@@ -289,10 +289,25 @@ def _enforce_tensor_spec(
     Enforce the input tensor shape and type matches the provided tensor spec.
     """
     expected_shape = tensor_spec.shape
+    expected_type = tensor_spec.type
     actual_shape = values.shape
-
     actual_type = values.dtype if isinstance(values, np.ndarray) else values.data.dtype
 
+    # This logic is for handling "ragged" arrays. The first check is for a standard numpy shape
+    # representation of a ragged array. The second is for handling a more manual specification
+    # of shape while support an input which is a ragged array.
+    if len(expected_shape) == 1 and expected_shape[0] == -1 and expected_type == np.dtype("O"):
+        # Sample spec: Tensor('object', (-1,))
```

Comment:
Can we also add an example of an input that would pass validation against this spec?